pkg/webhook/admission/webhook.go

```diff
@@ -84,23 +80,30 @@ type Webhook struct {
 }
 
 func (w *Webhook) setDefaults() {
-	if len(w.Name) == 0 {
-		w.Name = "admissionwebhook.k8s.io"
-	}
 	if len(w.Path) == 0 {
+		if len(w.Rules) == 0 || len(w.Rules[0].Resources) == 0 {
```

Comment:
should this logic be extracted to a helper or something?  Can the builder rely on the underlying webhook implementation to do the defaulting?

Reply:
Either way works. No strong opinion which way to use. @DirectXMan12 WDYT?

Reply:
> Can the builder rely on the underlying webhook implementation to do the defaulting?

This will make the builder return an incomplete webhook struct. But I guess it's not a big problem.

Reply:
I have dropped the defaulting in builder and only relying the defaulting in `webhook.go`